genie-web/src/main/java/com/netflix/genie/web/util/JobExecutionModeSelector.java

```diff
@@ -0,0 +1,233 @@
+    private static final String FORCE_AGENT_EXECUTION_HEADER_NAME = "genie-force-agent-execution";
+    private static final String FORCE_EMBEDDED_EXECUTION_HEADER_NAME = "genie-force-embedded-execution";
+    private static final String GLOBAL_AGENT_OVERRIDE_PROPERTY = PROPERTY_PREFIX + "force-agent";
+    private static final String GLOBAL_EMBEDDED_OVERRIDE_PROPERTY = PROPERTY_PREFIX + "force-embedded";
+    private static final String AGENT_PROBABILITY_PROPERTY = PROPERTY_PREFIX + "agent-probability";
+    private static final String METRIC_NAME = "genie.jobs.executionMode.counter";
+    private static final String OUTCOME_METRIC_TAG_NAME = "executeWithAgent";
+    private static final String CHECK_METRIC_TAG_NAME = "decidingCheck";
+
+    /**
+     * Property controlling default/fallback execution mode if none of the checks decides otherwise.
+     */
+    @VisibleForTesting
+    public static final String DEFAULT_EXECUTION_PROPERTY = PROPERTY_PREFIX + "default";
+
+    private final Environment environment;
+    private final Random random;
+    private final MeterRegistry meterRegistry;
+    private final List<Pair<String, CheckRequestFunction>> checks = ImmutableList.of(
+        // Global override that applies to all incoming jobs
+        new ImmutablePair<>("global-override", this::checkGlobalOverride),
+        // Job has HTTP header explicitly requesting a given execution mode
+        new ImmutablePair<>("header-override", this::checkForcingHeader),
+        // Job execution via agent for a fixed percentage of jobs, chosen randomly
+        new ImmutablePair<>("percentage", this::checkProbability)
+    );
+
+    /**
+     * Constructor.
+     *
+     * @param environment   the environment
+     * @param meterRegistry the meter registry
+     */
+    public JobExecutionModeSelector(
+        final Environment environment,
+        final MeterRegistry meterRegistry
+    ) {
+        this(new Random(), environment, meterRegistry);
+    }
+
+    /**
+     * Constructor with random generator used for tests.
+     *
+     * @param random        the random number generator
+     * @param environment   the environment
+     * @param meterRegistry the meter registry
+     */
+    @VisibleForTesting
+    JobExecutionModeSelector(
+        final Random random,
+        final Environment environment,
+        final MeterRegistry meterRegistry
+    ) {
+        this.environment = environment;
+        this.random = random;
+        this.meterRegistry = meterRegistry;
+    }
+
+
+    /**
+     * Inspect the incoming request and decide whether to execute using the new Agent-based execution mode.
+     *
+     * @param jobRequest         the job request
+     * @param httpServletRequest the raw http request
+     * @return true if this job should be executed using the agent
+     */
+    public boolean executeWithAgent(final JobRequest jobRequest, final HttpServletRequest httpServletRequest) {
```

Comment:
It might be preferable to decouple this from the communication technology (Servlet) and just pass in the bits needed to make the decision. However as this is a temporary migration class it's probably not a huge deal.